### ISSUE TITLE
`cri-o configured with cgroupfs cgroup manager, but received systemd slice as parent`

### ISSUE THREAD

**Issue by felipecrs** [nestybox/sysbox#893]
Tried installing sysbox 0.6.5 on a fresh cluster, and after deploying the manifest, now pods no longer work in the cluster.

Describing the pods I can see this issue:

```
Warning  FailedCreatePodSandBox  2s (x17 over 3m25s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = cri-o configured with cgroupfs cgroup manager, but received systemd slice as parent: /kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod368236bb_a128_48e2_a105_0b2c1c7a24e5.slice
```

It's a vanilla Kubernetes 1.28 cluster with Ubuntu 22.04 nodes, kernel 5.15.

**Comment by felipecrs** [comment]
My workaround ATM is to run the following in each node:

```
if grep 'cgroup_manager = "cgroupfs"' /etc/crio/crio.conf; then sudo sed -i 's,cgroup_manager = "cgroupfs",cgroup_manager = "systemd",' /etc/crio/crio.conf && sudo systemctl restart crio && grep 'cgroup_manager = "systemd"' /etc/crio/crio.conf; fi"
```

**Comment by ctalledo** [comment]
Hi @felipecrs, thanks for reporting and apologies for the belated reply.

Strange that you see this problem.

Here is the relevant code in sysbox-deploy-k8s that tries to configure both CRI-O to use the same cgroup manager as the Kubelet:
https://github.com/nestybox/sysbox-pkgr/blob/master/k8s/scripts/kubelet-config-helper.sh#L861

The code first gets the Kubelet cgroup manager, and then checks if CRI-O is using a different one. In particular, if the Kubelet is using the cgroupfs manager, the code reconfigures CRI-O to use cgroupfs. 

On the other hand, if the Kubelet is using the systemd manager, the leaves the CRI-O config untouched because presumably CRI-O defaults to `cgroup_manager = systemd` per [this CRI-O doc](https://github.com/cri-o/cri-o/blob/main/docs/crio.conf.5.md)).

But it looks like that latter part is where the bug resides. Instead of assuming CRI-O cgroup_manager is configured to use systemd, the code should explicitly configure it to use systemd (just like the work-around you provided above does).

I've not been able to repro, but if you want I can fix it and give you a development sysbox-deploy-k8s image to see if it works.

Let me know.

Thanks!

**Comment by felipecrs** [comment]
> I've not been able to repro yet, but we can add some logging in there to see try to debug it, and I can give you a development sysbox-deploy-k8s image to try.

Yes, if you provide this image with additional logging, I will run re-do the test and provide such additional information.
